### PR TITLE
Update JSON options examples

### LINKED ISSUE #26445: .NET 7 update required: JSON options configuration
[JSON options configuration](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-net-7-preview-1/#json-options-configuration)

`ConfigureRouteHandlerJsonOptions` has been renamed `ConfigureHttpJsonOptions`, see https://github.com/dotnet/aspnetcore/pull/42747

Need to update the following code for next release:

```
using System.Text.Json.Serialization;

var builder = WebApplication.CreateBuilder(args);

builder.Services.ConfigureRouteHandlerJsonOptions(options =>
{
    options.SerializerOptions.ReferenceHandler = ReferenceHandler.IgnoreCycles;
    options.SerializerOptions.AllowTrailingCommas = true;
    options.SerializerOptions.MaxDepth = 8;
});

var app = builder.Build();

app.MapGet("/", () => "Hello World!");

app.Run();
```

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #26445 

***@Rick-Anderson EDIT:***
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?view=aspnetcore-7.0&branch=pr-en-us-26996&tabs=visual-studio)

### CODE REVIEW

File: aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+#endregion
+#else
+#region snippet_2
+using System.Text.Json;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+var options = new JsonSerializerOptions(JsonSerializerDefaults.Web);
```
Comment (tdykstra) [inline aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs:34]: @captainsafia According to our [System.Text.Json](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-6-0#case-insensitive-deserialization) documentation, ASP.NET Core apps automatically get web defaults, so the use of options explicitly set to web defaults actually has no effect. Should we keep it as-is or set other options to make it a more realistic example?

Comment (captainsafia) [reply aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs:34]: Good catch! Let's try this so it's more realistic:

```suggestion
var options = new JsonSerializerOptions
{
  PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
  WriteIndented = true
}
```

File: aspnetcore/tutorials/min-web-api.md [hunk 2]
```diff
@@ -927,15 +927,18 @@ Verify you can't post or get the secret field.
...
+## Configure JSON serialization options
 
-[!code-csharp[](min-web-api/samples/6.x/WebMinJson/Program.cs?name=snippet_1)]
+The following example invokes `ConfigureHttpJsonOptions` to configure options that apply wherever the app serializes or deserializes JSON for HTTP requests and responses:
 
-The following code uses <xref:System.Text.Json.JsonSerializerOptions>:
+[!code-csharp[](min-web-api/samples/7.x/WebMinJson/Program.cs?name=snippet_1)]
 
-[!code-csharp[](min-web-api/samples/6.x/WebMinJson/Program.cs?name=snippet_2)]
+Options that you specify by using `ConfigureHttpJsonOptions` will apply whenever your app calls extension methods defined in <xref:Microsoft.AspNetCore.Http.HttpResponseJsonExtensions> and <xref:Microsoft.AspNetCore.Http.HttpRequestJsonExtensions>.
```
Comment (tdykstra) [inline aspnetcore/tutorials/min-web-api.md:937]: @captainsafia The name change from ConfigureRouteHandlerJsonOptions to ConfigureHttpJsonOptions doesn't seem to be in the latest preview yet.

File: aspnetcore/tutorials/min-web-api.md [hunk 3]
```diff
@@ -927,15 +927,18 @@ Verify you can't post or get the secret field.
...
+
+To make more localized changes to the serialization options, you can pass modified versions of <xref:System.Text.Json.JsonSerializerOptions> directly into the responses that are being sent from your endpoint, as shown in the following example:
+
+[!code-csharp[](min-web-api/samples/7.x/WebMinJson/Program.cs?name=snippet_2)]
 
 The preceding code uses [web defaults](/dotnet/standard/serialization/system-text-json-configure-options#web-defaults-for-jsonserializeroptions), which converts property names to camel case.
```
Comment (tdykstra) [inline aspnetcore/tutorials/min-web-api.md:943]:
```suggestion
```

File: aspnetcore/tutorials/min-web-api.md (hunk 2 above)
Comment (captainsafia) [reply aspnetcore/tutorials/min-web-api.md:937]: @tdykstra They should be in the RC1 that got release today.

Comment (tdykstra) [reply aspnetcore/tutorials/min-web-api.md:937]: Confirmed, thanks.

**Review by captainsafia — APPROVED** [review]

File: aspnetcore/tutorials/min-web-api.md [hunk 4]
```diff
@@ -927,17 +927,19 @@ Verify you can't post or get the secret field.
...
+## Configure JSON serialization options
 
-[!code-csharp[](min-web-api/samples/6.x/WebMinJson/Program.cs?name=snippet_1)]
+The following example invokes `ConfigureHttpJsonOptions` to configure options that apply wherever the app serializes or deserializes JSON for HTTP requests and responses:
 
-The following code uses <xref:System.Text.Json.JsonSerializerOptions>:
+[!code-csharp[](min-web-api/samples/7.x/WebMinJson/Program.cs?name=snippet_1)]
 
-[!code-csharp[](min-web-api/samples/6.x/WebMinJson/Program.cs?name=snippet_2)]
+Options that you specify by using `ConfigureHttpJsonOptions` will apply whenever your app calls extension methods defined in <xref:Microsoft.AspNetCore.Http.HttpResponseJsonExtensions> and <xref:Microsoft.AspNetCore.Http.HttpRequestJsonExtensions>.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/min-web-api.md:937]:
```suggestion
`ConfigureHttpJsonOptions` options apply when the app calls extension methods defined in <xref:Microsoft.AspNetCore.Http.HttpResponseJsonExtensions> or <xref:Microsoft.AspNetCore.Http.HttpRequestJsonExtensions>.
```

File: aspnetcore/tutorials/min-web-api.md [hunk 5]
```diff
@@ -927,17 +927,19 @@ Verify you can't post or get the secret field.
...
+
+To make more localized changes to the serialization options, you can pass modified versions of <xref:System.Text.Json.JsonSerializerOptions> directly into the responses that are being sent from your endpoint, as shown in the following example:
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/min-web-api.md:939]:
```suggestion
To make more localized changes to the serialization options, pass modified versions of <xref:System.Text.Json.JsonSerializerOptions> directly into the responses that are being sent from the endpoint, as shown in the following example:
```

File: aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
+#define First
+#if First
+// <snippet_1>
+var builder = WebApplication.CreateBuilder(args);
+
+// Configure JSON options
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs:6]: I'd remove that comment, the code is easy to read

File: aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+var app = builder.Build();
+
+app.MapGet("/", () => new Todo { Name = "Walk dog", IsComplete = false });
+
+app.Run();
+
+class Todo
+{
+    // These are public fields instead of properties.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/min-web-api/samples/7.x/WebMinJson/Program.cs:20]: Do we need that comment?

**Review by Rick-Anderson — APPROVED** [review]
Consider my changes. Prefer not using `you, your, etc`.

File: aspnetcore/tutorials/min-web-api.md [hunk 8]
```diff
@@ -927,17 +927,19 @@ Verify you can't post or get the secret field.
 - No support for [JsonPatch](https://www.nuget.org/packages/Microsoft.AspNetCore.JsonPatch/)
 - No support for [OData](https://www.nuget.org/packages/Microsoft.AspNetCore.OData/)
 - No support for [ApiVersioning](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Versioning/). See [this issue](https://github.com/dotnet/aspnet-api-versioning/issues/751) for more details.
-## Use JsonOptions
 
-The following code uses <xref:Microsoft.AspNetCore.Http.Json.JsonOptions>:
+## Configure JSON serialization options
 
-[!code-csharp[](min-web-api/samples/6.x/WebMinJson/Program.cs?name=snippet_1)]
+The following example invokes `ConfigureHttpJsonOptions` to configure options that apply wherever the app serializes or deserializes JSON for HTTP requests and responses:
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/tutorials/min-web-api.md:933]: I put in a work order to generate the RC1 APIs, but until then use source.dot.net

**Pull request merged**